Document/0x05b-Basic-Security_Testing.md

```diff
@@ -74,10 +74,91 @@ As mentioned before, starting with Android 7, the Android OS will no longer trus
+
+##### Adding the User Certificates to the NSC Configuration
+
+The CA certificates trusted by the app can be a system trusted CA as well as a user CA. If you added the certificate of your interception proxy already as additional CA in Android, you can force the Android app to trust it with the following NSC configuration below:
+
+```xml
+<network-security-config>
+   <base-config>
+      <trust-anchors>
+          <certificates src="system" />
+          <certificates src="user" />
+      </trust-anchors>
+   </base-config>
+</network-security-config>
+```
+
+To implement this new setting you must follow the steps below:
+
+- Decompile the app using decompilation tools.[Manual static Analysis](https://github.com/OWASP/owasp-mstg/blob/master/Document/0x05b-Basic-Security_Testing.md#manual-static-analysis) provides details about decompiling
+- Make the application trust user's certificates by creating a network security configuration that includes `<certificates src="user" />` as explained above
+- Repackage the app. The [Android developer documentation](https://developer.android.com/studio/publish/app-signing#signing-manually) explains how it's done.
+
+Note that even if this method is quite simple its major drawback is that you have to apply this operation for each application you want to evaluate which is additional overhead for testing.
+
+There is a python script available that automates the steps described above called [Android-CertKiller](https://github.com/51j0/Android-CertKiller "Android-CertKiller"). This Python script can extract the APK from an installed Android app, decompile it, make it debuggable, add a new network security config that allows user certificates, builds and signs the new APK and installs the new APK with the SSL Bypass. The last step, [installing the app might fail](https://github.com/51j0/Android-CertKiller/issues "APK not installing"), due to a bug at the moment.  
+
+```bash
+python main.py -w
+
+***************************************
+Android CertKiller (v0.1)
+***************************************
+
+CertKiller Wizard Mode
+---------------------------------
+List of devices attached
+4200dc72f27bc44d	device
+
+---------------------------------
+
+Enter Application Package Name: nsc.android.mstg.owasp.org.android_nsc
+
+Package: /data/app/nsc.android.mstg.owasp.org.android_nsc-1/base.apk
+
+I. Intitaing APK extraction from device
+   complete
+------------------------------
+I. Decompiling
+   complete
+------------------------------
+I. Applying SSL bypass
+   complete
+------------------------------
+I. Building New APK
+   complete
+------------------------------
+I. Sigining APK
+   complete
+------------------------------
+
+Would you like to install the APK on your device(y/N): y
+------------------------------------
+ Installing Unpinned APK
+------------------------------
+Finished
+```
+
+
+##### Adding the Proxy Certificate to the NSC Configuration
```

Comment:
Why exactly do we want this approach? Are there pro's vs adding it to the user store?

Reply:
This was existing content. I just got rid of it and changed the user setting explaining all available settings.